lib/clementine_server.js

```diff
@@ -0,0 +1,222 @@
+var util = require('util');
+var ClementineServer = require('clementine-remote').Server;
+var Player = require('./player');
+var log = require('./log');
+
+function ClementineApiServer(player) {
+  var self = this;
+
+  this.player = player;
+  this.server = null;
+
+  /*player.on('repeatUpdate', updateOptionsSubsystem);
+  player.on('autoDjOn', updateOptionsSubsystem);
+  player.on('queueUpdate', onQueueUpdate);
+  player.on('deleteDbTrack', updateDatabaseSubsystem);
+  player.on('updateDb', updateDatabaseSubsystem);
+  player.on('playlistCreate', updateStoredPlaylistSubsystem);
+  player.on('playlistUpdate', updateStoredPlaylistSubsystem);
+  player.on('playlistDelete', updateStoredPlaylistSubsystem);*/
+}
+
+ClementineApiServer.prototype.listen = function (port, host, done) {
+  var player = this.player;
+
+  var server = new ClementineServer({
+    host: host,
+    port: port
+  });
+
+  // Current position
+  // TODO: something more reliable
+  var position = 0, positionInterval;
```

Comment:
hmm this is not very well designed. I kind of want to suggest that we go bug clementine to support the groove basin protocol instead :-/